tools/ml-audio-content-moderation/README.md

```diff
@@ -0,0 +1,373 @@
+
+````
+export staging_audio_bucket=staging-audio-files-$STATIC_UUID
+
+gsutil mb gs://$staging_audio_bucket
+````
+
+````
+export processed_audio_bucket=processed-audio-files-$STATIC_UUID
+gsutil mb gs://$processed_audio_bucket
+````
+
+````
+export error_audio_bucket=error-audio-files-$STATIC_UUID
+gsutil mb gs://$error_audio_bucket
+````
+
+
+````
+export transcription_bucket=transcription-files-$STATIC_UUID
+gsutil mb gs://$transcription_bucket
+````
+
+````
+export output_bucket=output-files-$STATIC_UUID
+gsutil mb gs://$output_bucket
+````
+
+
+4. Deploy first Cloud Function to Send STT API 
+
+Change directories into the send_stt_api_function directory
+
+````
+cd send_stt_api_function/
+````
+
+Deploy function
+
+````
+gcloud functions deploy send_stt_api \
+  --entry-point main \
+  --runtime python37 \
+  --trigger-resource $staging_audio_bucket \
+  --trigger-event google.storage.object.finalize \
+  --timeout 540s \
+  --set-env-vars topic_name=$TOPIC_NAME,error_bucket=$error_audio_bucket
+````
+
+5. Deploy second Cloud Function to Read STT API Output
+
+Change directories into the read_stt_api_function
+
+````
+cd ../read_stt_api_function/
+````
+
+````
+gcloud functions deploy read_stt_api \
+  --entry-point main \
+   --runtime python37 \
+   --trigger-resource cron_topic \
+   --trigger-event google.pubsub.topic.publish \
+   --timeout 540s \
+   --set-env-vars topic_name=$TOPIC_NAME,subscription_name=$SUBSCRIPTION_NAME,transcription_bucket=$transcription_bucket,staging_audio_bucket=$staging_audio_bucket,processed_audio_bucket=$processed_audio_bucket,error_audio_bucket=$error_audio_bucket
+````
+
+6. Deploy Cloud Scheduler Job
+
+````
+gcloud scheduler jobs create pubsub check_stt_job \
+  --schedule "*/10 * * * *" \
+  --topic cron_topic \
+  --message-body "Check Speech-to-text results"
+````
+
+Note that you can edit the `schedule` flag to be any interval in UNIX cron. By default, our solution
+uses every 10 minutes.
+
+
+7. Deploy Perspective API Function
+
+Change directories into the perspective_api_function directory
+
+````
+cd ../perspective_api_function/
+````
+
+````
+gcloud functions deploy perspective_api \
+  --entry-point main \
+  --runtime python37 \
+  --trigger-resource $transcription_bucket \
+  --trigger-event google.storage.object.finalize \
+  --timeout 540s \
+  --set-env-vars output_bucket=$output_bucket
+````
+
+8. Deploy NLP Function
+
+Change directories into the nlp_api_function directory
+
+````
+cd ../nlp_api_function/
+````
+
+````
+gcloud functions deploy nlp_api \
+  --entry-point main \
+  --runtime python37 \
+  --trigger-resource $transcription_bucket \
+  --trigger-event google.storage.object.finalize \
+  --timeout 540s \
+  --set-env-vars output_bucket=$output_bucket
+````
+
+
+### Step 2b
+
+1. [Install Terraform](https://learn.hashicorp.com/terraform/getting-started/install)
+
+2. Edit `variables.tf` with all of your associated variables. You must edit the project_id, but the 
```

Comment:
Users shouldn't directly edit the variables file. Instead, they can create a terraform.tfvars file to customize: https://learn.hashicorp.com/terraform/getting-started/variables.html#from-a-file